spec/client_spec.rb

```diff
@@ -86,13 +95,56 @@
         :at => (Time.now.to_i - 100) * 1000
       }
     end
+    let(:presence6) do
+      {
+        :userId => user_id6,
+        :userType => 2,
+        :userData => 'userData6',
+        :clientId => client_id6,
+        :online => true,
+        :at => Time.now.to_i * 1000,
+        :sentry => sentry_id2
+      }
+    end
+    let(:presence7) do
+      {
+        :userId => user_id6,
+        :userType => 2,
+        :userData => 'userData7',
+        :clientId => client_id7,
+        :online => true,
+        :at => Time.now.to_i * 1000,
+        :sentry => sentry_id3
+      }
+    end
+    let(:sentry1) do
+      {
+        :name => sentry_id1,
+        :expiration => (Time.now.to_i + 100) * 1000,
+        :host => "precise64",
+        :port => "8000"
+      }
+    end
+    let(:sentry2) do
+      {
+        :name => sentry_id2,
+        :expiration => (Time.now.to_i - 100) * 1000,
+        :host => "precise64",
+        :port => "8000"
+      }
+    end
 
     before do
       fakeredis.hset(key, "#{user_id1}.#{client_id1}", presence1.to_json)
       fakeredis.hset(key, "#{user_id2}.#{client_id2}", presence2.to_json)
       fakeredis.hset(key, "#{user_id1}.#{client_id3}", presence3.to_json)
       fakeredis.hset(key, "#{user_id4}.#{client_id4}", presence4.to_json) # offline
-      fakeredis.hset(key, "#{user_id5}.#{client_id5}", presence5.to_json) # timeout
```

Comment:
I removed this because we hadn't implemented this case. I also noticed that presence messages in my local machine don't have the `at` key. So perhaps this is no longer relevant?